src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -1308,6 +1362,20 @@ public void CreateModel(XmlElement modelData)
             {
                 var noteModel = NodeGraph.LoadNoteFromXml(modelData);
                 Notes.Add(noteModel);
+
+                //check whether this note belongs to a group
+                foreach (var annotation in Annotations)
+                {
+                    //this note "was" in a group
+                    if (annotation.DeletedModelBases.Any(m => m.GUID == noteModel.GUID))
+                    {
+                        var list = annotation.SelectedModels.ToList();
+                        list.Add(noteModel);
+                        annotation.SelectedModels = list;
+                        annotation.loadFromXML = false;
+                        annotation.UpdateBoundaryFromSelection();
```

Comment:
Hmm, why not just add a method to `AnnotationModel` that's something like `AddToSelectedModels`.  What you're doing here is repeatedly copying the lists for every `AnnotationModel`.